base/methodshow.jl

```diff
@@ -266,12 +284,31 @@ function show_method_table(io::IO, ms::MethodList, max::Int=-1, header::Bool=tru
     last_shown_line_infos = get(io, :last_shown_line_infos, nothing)
     last_shown_line_infos === nothing || empty!(last_shown_line_infos)
 
+    modul = if mt === _TYPE_NAME.mt  # type constructor
+            which(ms.ms[1].module, ms.ms[1].name)
+        else
+            mt.module
+        end
+    modulecolordict = Dict{Module, Symbol}()
+    modulecolorcycler = Iterators.Stateful(Iterators.cycle(METHODLIST_MODULECOLORS))
+    modulecolordict[parentmodule_before_main(modul)] = :blue
```

Comment:
Is this really our only strategy for this? Global state through a particular list?

Reply:
I have no memory of writing this :/  I presume I tried to copy the stack traces.

The list [:cyan, :green, :yellow] is global, but isn't `modulecolorcycler` local to this function?

Reply:
ah, true, seems to be a copy from there. I guess this is okay